eng/scripts/check-external-dependency.ps1

```diff
@@ -0,0 +1,100 @@
+[CmdletBinding(SupportsShouldProcess = $true)]
+param (
+  [Parameter(Mandatory = $true)]
+  [string]$AuthToken,
+
+  [Parameter(Mandatory = $true)]
+  [string]$RepoOwner,
+
+  [Parameter(Mandatory = $true)]
+  [string]$RepoName
+)
+
+
+$dependencyUpgradeLabel = "dependency-upgrade-required"
+$dependencyRegex = "^\+\s(?<pkg>[\S]*)\s(?<version>[\S]*)\s\((?<newVersion>[0-9\.a-b]*).*\)\s?(?<deprecated>deprecated)?"
+$RepoRoot = Resolve-Path -Path "${PSScriptRoot}/../.."
+$rushFile = join-Path -Path $RepoRoot "rush.json"
+$commonConfigFile = Join-path -Path $RepoRoot "common" "config" "rush" "common-versions.json"
+
+$EngCommonScriptsPath = Join-Path (Resolve-Path "${PSScriptRoot}/..") "common" "scripts"
+Test-Path $EngCommonScriptsPath
+. (Join-Path $EngCommonScriptsPath common.ps1)
+
+
+# Check and return if an isue already exists to upgrade the package 
+function Get-GithubIssue($PackageName, $IsDeprecated) {
+  $issueTitle = "Dependency package $PackageName has a new version available"
+  $issues = Get-GitHubIssues -RepoOwner $RepoOwner -RepoName $RepoName -CreatedBy "praveenkuttappan" -Labels "dependency-upgrade-required" -AuthToken $AuthToken
+  if ($issues) {
+    foreach ($issue in $issues) {
+      if ($issue.title -eq $issueTitle) {
+        return $issue
+      }
+    }
+  }
+  return $issues
+}
+
+# Create new issue if none exists for the package. or update current one if an issue exists
+function Set-GitHubIssue($Package) {
+  $pkgName = $Package.Name
+  $issueTitle = "Dependency package $pkgName has a new version available"
+  $issueDesc = "We have identified a dependency on $pkgName ($($Package.OldVersion)). "
+
+  if ($Package.IsDeprecated) {
+    $issueDesc = "Version $($Package.OldVersion) of $pkgName has been deprecated.`n"
+  }
+  $issueDesc += "A new version ($($Package.NewVersion) is available now."
+
+
+  $issue = Get-GithubIssue -PackageName $Package.Name -IsDeprecated $Package.IsDeprecated
+  if ($issue) {
+    Update-GitHubIssue -RepoOwner $RepoOwner -RepoName $RepoName -AuthToken $AuthToken -IssueNumber $issue.number -Body $issueDesc
+  }
+  else {
+    $newIssue = New-GitHubIssue -RepoOwner $RepoOwner -RepoName $RepoName -AuthToken $AuthToken -Title $issueTitle -Description $issueDesc  
+    Write-Host "Created issue $($newIssue.number) with title '$issueTitle'"
+    if ($newIssue) {
+      $out = Add-GitHubIssueLabels -RepoOwner $RepoOwner -RepoName $RepoName -AuthToken $AuthToken -Labels $dependencyUpgradeLabel -IssueNumber $newIssue.number
+      Write-Host "Addd label $dependencyUpgradeLabel to issue $($newIssue.number)"
+    }
+  }
+}
+
+
+# Update rush configuration files to alter settings
+if ((Test-Path $rushFile) -and (Test-Path $commonConfigFile)) {
+  $rushJson = Get-Content -Path $rushFile | ConvertFrom-Json
```

Comment:
Based on the name of this script I wouldn't have expected it to modify the rush configurations. Are these supposed to be temporary changes to get the correct output we want? If so, we might want to try and undo them once the script is done so we don't leave the repo in a dirty state.

Reply:
These are temporary config change. I will revert them post rush update.